### PR TITLE
Basic DNS fallback

### PR DESCRIPTION
**Pull request by yzqzss**

### CODE REVIEW

File: dns.go [hunk 1]
```diff
@@ -41,20 +43,34 @@ func (d *customDialer) archiveDNS(address string) (resolvedIP net.IP, err error)
...
+
+	fallbackServers := min(maxFallbackDNSServers, len(d.DNSConfig.Servers)-1)
 
-	go func() {
-		defer wg.Done()
-		ipv4, errA = d.lookupIP(address, dns.TypeA)
-	}()
+	for DNSServer := 0; DNSServer <= fallbackServers; DNSServer++ {
+		if DNSServer > 0 {
+			// TODO: print warning if the default DNS server is not reachable
```
Comment (yzqzss) [inline dns.go:54]: I don't know how to log the warning here properly. Add a fmt or slog print? :(

**Comment by yzqzss** [comment]
~~Thank you for not implementing the DNS Fallback mechanism yet.~~
This helped me accidentally discover the primary DNS unreachable issue of my server /etc/resolv.conf configuration, and the various network problems (e.g. SMTP timeout) that have troubled me for a year have been solved!

LMAO 😂😂

File: dns.go (hunk 1 above)
Comment (CorentinB) [reply dns.go:54]: I guess slog is better but it will eventually get rewritten when we properly rewrite the logging..

**Comment by CorentinB** [comment]
> ~Thank you for not implementing the DNS Fallback mechanism yet.~ This helped me accidentally discover the primary DNS unreachable issue of my server /etc/resolv.conf configuration, and the various network problems (e.g. SMTP timeout) that have troubled me for a year have been solved!
> 
> LOL 😂😂

AH! 😂

**Pull request merged**